bootstrap/bootstrapper/fs/source.go

```diff
@@ -88,18 +71,25 @@ func (fss *fileSystemSource) GetAvailability(shard uint32, targetRangesForShard
 
 // ReadData returns raw series for a given shard within certain time ranges.
 func (fss *fileSystemSource) ReadData(shard uint32, tr xtime.Ranges) (m3db.ShardResult, xtime.Ranges) {
-	log := fss.opts.GetLogger()
-
 	if xtime.IsEmpty(tr) {
 		return nil, tr
 	}
-	files, err := fss.getInfoFiles(shard)
-	if err != nil {
-		log.Errorf("unable to get info files for shard %d: %v", shard, err)
+
+	var files []string
+	fs.ForEachInfoFile(
```

Comment:
So right now the ranges passed to the `ReadData` are those returned from `GetAvailability`, but I'd rather not rely on that implicit assumption and prefer to revalidate independently. Also the time ranges passed in might not be aligned on block size boundaries (e.g., if there is a bootstrapper that takes higher precedence than the filesystem one), in which case we'll need to scan the info files anyway.